swap_meet/vendor.py

```diff
@@ -1,2 +1,72 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = [item for item in self.inventory if item.category == category]
+        return items_by_category
+
+    def swap_items(self, other, yours, theirs):
+        if yours in set(self.inventory) and theirs in set(other.inventory):
+            other.add(yours)
+            self.remove(yours)
+            self.add(theirs)
+            other.remove(theirs)
+            return True
+
+    def swap_first_item(self, other):
+        try:
+            yours = self.inventory[0]   
+            theirs = other.inventory[0]
+            other.add(yours)
+            self.add(theirs)
+            self.remove(yours)
+            other.remove(theirs)
+            return True
+        except IndexError:
+            return False
+
+    def get_best_by_category(self, category):
+        try:
+            category_list = self.get_by_category(category)
+            best = max(category_list, key=lambda x: x.condition)
```

Comment:
Nice use of max with a key function for comparing the conditions. We should know everything we need to know in order to fully explain what this is doing. Try writing your own version of the max() function that can also be optionally customized with a key function1